src/commands/dev/mod.rs

```diff
@@ -16,21 +17,24 @@ pub fn dev(
     target: Target,
     deploy_config: DeployConfig,
     user: Option<GlobalUser>,
-    host: Option<&str>,
-    port: Option<u16>,
-    ip: Option<&str>,
+    server_config: ServerConfig,
+    http: bool,
     verbose: bool,
 ) -> Result<(), failure::Error> {
-    let server_config = ServerConfig::new(host, ip, port)?;
-
     // before serving requests we must first build the Worker
     build(&target)?;
 
     match user {
         // authenticated users connect to the edge
-        Some(user) => edge::dev(target, user, server_config, deploy_config, verbose),
+        Some(user) => edge::dev(target, user, server_config, deploy_config, http, verbose),
 
         // unauthenticated users connect to gcs
-        None => gcs::dev(target, server_config, verbose),
+        None => {
+            if http {
+                failure::bail!("Unauthenticated dev must use https")
```

Comment:
i dont think this should be true - though we should figure out how `--host` and `--http` interact. my guess is that if the protocol mismatch in the two args then it should fail but other than that it should act the same as edge dev